e2e/actions/actions.go

```diff
@@ -1017,6 +1017,358 @@ func (c actionTests) actionNetwork(t *testing.T) {
+			},
+			postRun: checkHostDir(filepath.Join(hostHomeDir, "canary121RO")),
+			exit:    0,
+		},
+		{
+			name: "CustomHomeBind",
+			args: []string{
+				"--home", hostHomeDir + ":/home/e2e",
+				"--bind", hostCanaryDir + ":/home/e2e/canaryRO",
+				sandbox,
+				"test", "-f", "/home/e2e/canaryRO/file",
+			},
+			postRun: checkHostDir(filepath.Join(hostHomeDir, "canaryRO")),
+			exit:    0,
+		},
+		{
+			name: "CustomHomeBindWritableOK",
+			args: []string{
+				"--home", hostHomeDir + ":/home/e2e",
+				"--bind", hostCanaryDir + ":/home/e2e/canaryRW",
+				"--writable",
+				sandbox,
+				"test", "-f", "/home/e2e/canaryRW/file",
+			},
+			postRun: checkHostDir(filepath.Join(hostHomeDir, "canaryRW")),
+			exit:    0,
+		},
+		{
+			name: "CustomHomeBindWritableKO",
+			args: []string{
+				"--home", canaryDirBind,
+				"--writable",
+				sandbox,
+				"true",
+			},
+			exit: 255,
+		},
+		{
+			name: "WorkdirTmpBind",
+			args: []string{
+				"--workdir", hostWorkDir,
+				"--contain",
+				"--bind", hostCanaryDir + ":/tmp/canary/dir",
+				sandbox,
+				"test", "-f", "/tmp/canary/dir/file",
+			},
+			postRun: checkHostDir(filepath.Join(hostWorkDir, "tmp", "canary/dir")),
+			exit:    0,
+		},
+		{
+			name: "WorkdirTmpBindWritable",
+			args: []string{
+				"--writable",
+				"--workdir", hostWorkDir,
+				"--contain",
+				"--bind", hostCanaryDir + ":/tmp/canary/dir",
+				sandbox,
+				"test", "-f", "/tmp/canary/dir/file",
+			},
+			postRun: checkHostDir(filepath.Join(hostWorkDir, "tmp", "canary/dir")),
+			exit:    0,
+		},
+		{
+			name: "WorkdirVarTmpBind",
+			args: []string{
+				"--workdir", hostWorkDir,
+				"--contain",
+				"--bind", hostCanaryDir + ":/var/tmp/canary/dir",
+				sandbox,
+				"test", "-f", "/var/tmp/canary/dir/file",
+			},
+			postRun: checkHostDir(filepath.Join(hostWorkDir, "var_tmp", "canary/dir")),
+			exit:    0,
+		},
+		{
+			name: "WorkdirVarTmpBindWritable",
+			args: []string{
+				"--writable",
+				"--workdir", hostWorkDir,
+				"--contain",
+				"--bind", hostCanaryDir + ":/var/tmp/canary/dir",
+				sandbox,
+				"test", "-f", "/var/tmp/canary/dir/file",
+			},
+			postRun: checkHostDir(filepath.Join(hostWorkDir, "var_tmp", "canary/dir")),
+			exit:    0,
+		},
+		{
+			name: "ScratchTmpfsBind",
+			args: []string{
+				"--scratch", "/scratch",
+				"--bind", hostCanaryDir + ":/scratch/dir",
+				sandbox,
+				"test", "-f", "/scratch/dir/file",
+			},
+			exit: 0,
+		},
+		{
+			name: "ScratchWorkdirBind",
+			args: []string{
+				"--workdir", hostWorkDir,
+				"--scratch", "/scratch",
+				"--bind", hostCanaryDir + ":/scratch/dir",
+				sandbox,
+				"test", "-f", "/scratch/dir/file",
+			},
+			postRun: checkHostDir(filepath.Join(hostWorkDir, "scratch/scratch", "dir")),
+			exit:    0,
+		},
+	}
+
+	for _, profile := range e2e.Profiles {
+		profile := profile
+		createWorkspaceDirs(t)
+
+		t.Run(profile.String(), func(t *testing.T) {
```

Comment:
Is the `t.Run` necessary here? I thought `c.env.RunSingularity()` will take care of the `t.Run` stuff?

Reply:
Yes `RunSingularity` could do that too, here this is just a convenient way to execute all tests for a profile within a parent test with profile name.